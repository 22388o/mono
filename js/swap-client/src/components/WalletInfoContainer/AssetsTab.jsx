import { useState } from "react";
import { Grid, Stack, Typography } from "@mui/material"

import styles from "../../styles/WalletInfoContainer.module.css";
import { CoinsSubTab } from "./CoinsSubTab";

export const AssetsTab = () => {
  const [activeTab, setActiveTab] = useState('coins');

  return (
    <>
      <Stack direction='row' sx={{justifyContent:'center', display:'flex'}} gap={1}>
        <Typography sx={{color: 'white', fontSize: '32px', fontFamily: 'NotoBold'}}>1293.00</Typography>
        <Typography sx={{color: '#6A6A6A', fontSize: '16px', marginTop: '18px'}}>USD</Typography>
      </Stack>
      <Grid container sx={{marginTop: '30px'}}>
        <Grid item xs={6}>
          <Typography className={[
              styles['sub-tab-item'], 
              activeTab === 'coins' ? styles['left-sub-tab-active'] : styles['sub-tab-inactive']
            ]}
            onClick={() => setActiveTab('coins')}
          >
            Coins
          </Typography>
        </Grid>
        <Grid item xs={6}>
          <Typography className={[
              styles['sub-tab-item'], 
              activeTab === 'coll' ? styles['right-sub-tab-active'] : styles['sub-tab-inactive']
            ]}
            onClick={() => setActiveTab('coll')}
          >
            Collectibles
          </Typography>
        </Grid>
      </Grid>
      <Grid className={styles['sub-container']}>
        { activeTab === 'coins' && <CoinsSubTab />}
      </Grid>
    </>
  )
}