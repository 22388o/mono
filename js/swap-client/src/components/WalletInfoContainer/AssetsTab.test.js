import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // For the "toBeInTheDocument" matcher
import { AssetsTab } from './AssetsTab';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ data: { amount: 12 } }),
  })
);
describe('AssetsTab component test', () => {
  it('renders component', () => {

    const {container} = render(
      <AssetsTab />
    );

    expect(container).toMatchSnapshot();
  });
});
