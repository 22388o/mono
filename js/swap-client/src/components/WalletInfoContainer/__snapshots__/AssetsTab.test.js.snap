// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsTab component test renders component 1`] = `
<div>
  <div
    class="MuiStack-root css-18yyqbo-MuiStack-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-7ut9zg-MuiTypography-root"
    >
      NaN
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-fqcpc4-MuiTypography-root"
    >
      USD
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-544k9b-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 
              undefined 
              undefined
             css-ahj2mt-MuiTypography-root"
      >
        Coins
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 
              undefined 
              undefined
             css-ahj2mt-MuiTypography-root"
      >
        Collectibles
      </p>
    </div>
  </div>
  <div
    class="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <div
      class="MuiStack-root css-nen11g-MuiStack-root"
    >
      <div>
        <div
          class="MuiStack-root css-m69qwo-MuiStack-root"
        >
          <div
            class="MuiStack-root css-o389km-MuiStack-root"
          >
            <img
              src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
              width="32"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              BTC
            </p>
          </div>
          <div
            class="MuiStack-root css-1qxw92a-MuiStack-root"
          >
            <div
              class="MuiStack-root css-1qvzoxq-MuiStack-root"
            >
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
                >
                  BTC
                </p>
              </div>
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1b9ac31-MuiTypography-root"
                >
                  NaN
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ajxvt4-MuiTypography-root"
                >
                  USD
                </p>
              </div>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
      <div>
        <div
          class="MuiStack-root css-m69qwo-MuiStack-root"
        >
          <div
            class="MuiStack-root css-o389km-MuiStack-root"
          >
            <img
              src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
              width="32"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              ETH
            </p>
          </div>
          <div
            class="MuiStack-root css-1qxw92a-MuiStack-root"
          >
            <div
              class="MuiStack-root css-1qvzoxq-MuiStack-root"
            >
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
                >
                  ETH
                </p>
              </div>
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1b9ac31-MuiTypography-root"
                >
                  NaN
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ajxvt4-MuiTypography-root"
                >
                  USD
                </p>
              </div>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
      <div>
        <div
          class="MuiStack-root css-m69qwo-MuiStack-root"
        >
          <div
            class="MuiStack-root css-o389km-MuiStack-root"
          >
            <img
              src="https://seeklogo.com/images/T/tether-usdt-logo-FA55C7F397-seeklogo.com.png"
              width="32"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              USDT
            </p>
          </div>
          <div
            class="MuiStack-root css-1qxw92a-MuiStack-root"
          >
            <div
              class="MuiStack-root css-1qvzoxq-MuiStack-root"
            >
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
                >
                  1000
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
                >
                  USDT
                </p>
              </div>
              <div
                class="MuiStack-root css-7vetex-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1b9ac31-MuiTypography-root"
                >
                  1,000
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ajxvt4-MuiTypography-root"
                >
                  USD
                </p>
              </div>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
    </div>
  </div>
</div>
`;
